packages/build-tools/src/utils/appConfig.ts

```diff
@@ -1,17 +1,31 @@
 import { getConfig, ProjectConfig } from '@expo/config';
 import { Env } from '@expo/eas-build-job';
 import { bunyan, LoggerLevel } from '@expo/logger';
+import { load } from '@expo/env';
+import semver from 'semver';
 
-export function readAppConfig(projectDir: string, env: Env, logger: bunyan): ProjectConfig {
+export function readAppConfig({
+  projectDir,
+  env,
+  logger,
+  sdkVersion,
+}: {
+  projectDir: string;
+  env: Env;
+  logger: bunyan;
+  sdkVersion?: string;
+}): ProjectConfig {
   const originalProcessEnv: NodeJS.ProcessEnv = process.env;
   const originalProcessExit = process.exit;
   const originalProcessCwd = process.cwd;
   const originalStdoutWrite = process.stdout.write;
   const originalStderrWrite = process.stderr.write;
 
   const stdoutStore: { text: string; level: LoggerLevel }[] = [];
+  const shouldLoadEnvVarsFromDotenvFile = sdkVersion && semver.satisfies(sdkVersion, '>=49');
+  const envVarsFromDotenvFile = shouldLoadEnvVarsFromDotenvFile ? load(projectDir) : {};
   try {
-    process.env = { ...env };
+    process.env = { ...env, ...envVarsFromDotenvFile };
```

Comment:
This might have unexpected side effects in Bun, preferably we don’t overwrite `process.env` with an object. Instead, setting the properties from the env file to the existing object might be better here.

See https://github.com/expo/expo/pull/25393 for more info on this “mindf*ck” 🙈

Reply:
@byCedric do you mean something like the changes I added in my last commit?

Reply:
Yes, exactly 😄